network/p2p/subscription/subscriptionManager.go

```diff
@@ -11,15 +11,20 @@ import (
 // ChannelSubscriptionManager manages subscriptions of engines running on the node to channels.
 // Each channel should be taken by at most a single engine.
 type ChannelSubscriptionManager struct {
-	mu      sync.RWMutex
-	engines map[channels.Channel]network.MessageProcessor
-	mw      network.Middleware
+	mu         sync.RWMutex
+	engines    map[channels.Channel]network.MessageProcessor
+	middleware network.Underlay // the Underlay interface of the network layer
```

Comment:
```suggestion
	networkUnderlay network.Underlay // the Underlay interface of the network layer
```
Do we still need to use "middleware" at all?

Reply:
https://github.com/onflow/flow-go/pull/4664/commits/5c7a8d2d582814dd2ebdd166207bdd02ec843b97